### PR TITLE
fix(osmoutils/CL): bug in accumulator delete and add tor rewards (fungification)

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

This PR introduces fixes to the accumulator package. 

There were 2 problems:

1. `DeletePosition` would delete the position but not decrement the global accumulator. As a result , incentive claiming would break.

`DeletePosition` is not meant to be exposed. Instead, position deletion should happen only during reward claiming so that rewards are not missed and the global accumulator value is decremented.

2. `AddToUnclaimedRewards` would overwrite the postion's accumulator value with the global accumulator value.

This was fixed and tests added

## Brief Changelog

- fix delete positon bug
- unexport delete positon
- reorder logic in fungifyPosition to delete in claim rewards
- add to unclaimed rewards bug fix
- add tests for add to unclaimed rewards,

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`?  no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: x/concentrated-liquidity/position.go
Comment (p0mvn) [inline x/concentrated-liquidity/position.go:495]: Note: this is bug number 1 in the description where it wouldn't correctly decrement global accum shares.

File: osmoutils/accum/accum.go
Comment (p0mvn) [inline osmoutils/accum/accum.go:404]: Note: this is bug number 2 in description

File: osmoutils/accum/accum_test.go
Comment (mattverse) [inline osmoutils/accum/accum_test.go:1600]: would it be possible to add test case for position num shares being zero and having it deleted?

File: x/concentrated-liquidity/position.go
Comment (mattverse) [inline x/concentrated-liquidity/position.go:492]: good catch! Would it be possible to add tests cases that contians this change please?

**Review by mattverse — APPROVED** [review]
Both 1 and 2 fix LGTM, approving so that I don't block this PR, but requesting trivial test cases

File: osmoutils/accum/accum_test.go
Comment (p0mvn) [reply osmoutils/accum/accum_test.go:1604]: I'm unsure what you mean by this.

Assuming that a test case where the rewards added are 0 or empty.

Added here: [ec2754d](https://github.com/osmosis-labs/osmosis/pull/5057/commits/ec2754d64fec91a1cb8c7d2fb128b75fda6f276e)

File: x/concentrated-liquidity/position.go
Comment (p0mvn) [reply x/concentrated-liquidity/position.go:492]: So this is already covered by test cases in general.

I have this particular bug reproduced in a test case in another branch. Unfortunately, that test case depends on the change that I haven't yet merged in.

Ref: https://github.com/osmosis-labs/osmosis/issues/5038

I will PR this repro test as soon as I address #5038

**Pull request merged**
